### ISSUE TITLE
I have one more question.

### ISSUE THREAD

**Issue by rimiyeyo** [etzinis/unsup_speech_enh_adaptation#6]
Hi Thank you for reply. I ask the question again because it seems to be different from the intention of my question.

When I train the student model with the teacher model ckpt provided by you, it produces good results.
However, if I train the student model with the reproduced teacher model, trained by using the provided script, I get different results.

To sum up, the results are not reproducible.
Are there any additional code changes or did you train the teacher with more epochs?
If not, are there any specific metrics you look at the comet.ml site?

always thanks

**Comment by etzinis** [comment]
I have trained the teacher model with the script that I have in this repo, thus, all results should be reproducible.

But when you say the results are not reproducible, what are you doing different than my scripts and what are the differences in the final results?

**Comment by rimiyeyo** [comment]
Hello! 
1. When I tried to download the chime-5 dataset from the link you provided on your github (https://github.com/UDASE-CHiME2023/unlabeled_data), I got a "not found 404" link error. So I took a detour and cloned the CHiME-5 folder from https://github.com/UDASE-CHiME2023 link. I don't remember if it was the 5th Chime challenge or the 6th Chime challenge dataset, but I downloaded the dataset from one of the sites. I noticed that the json_files and challenge dataset in the cloned CHiME-5 folder are slightly different in format, and there is no eval folder. There are also no eval-related json_files. Still, I created a processed_CHiME from the code in the clone file and used it to train. I used the libri3mix datastet for everything else.

What Chime challenge dataset did you use, is it different from what the author used here?

2. I am curious about the optimal epoch subtraction method. When I save the model on the comet site, I get several metrics. I already know that the higher the MOS, the better, and the lower the SISDR, the better the training, but since SISDR is negative, the higher the better. And I wonder if it uses the higher value of the rest of the speech mos as an indicator, or if it uses the epoch where the noise mos is good. Also, it used to display the mean, std, and median together, and I wonder what the purpose of displaying the three indicators together is. 

I'm curious how the author checked the metrics to see which of the three metrics is meaningful, and did he consider mos or mean or std or median to choose the best epoch?

Thanks

**Comment by etzinis** [comment]
> When I tried to download the chime-5 dataset from the link you provided on your github (https://github.com/UDASE
CHiME2023/unlabeled_data), I got a "not found 404" link error.

This is not *my* repo and it should be https://github.com/UDASE-CHiME2023/CHiME-5, sorry about that.

>  I don't remember if it was the 5th Chime challenge or the 6th Chime challenge dataset, but I downloaded the dataset from one of the sites. 

Both datasets are the same 

> I noticed that the json_files and challenge dataset in the cloned CHiME-5 folder are slightly different in format, and there is no eval folder. There are also no eval-related json_files.

The eval folder is not yet available since the challenge is still running. Eval files will become available at the last few weeks before the deadline. The results I provide are also on the validation set.

> What Chime challenge dataset did you use, is it different from what the author used here?

Both are the same for our purposes. I do not understand who is the author and where is here.

> I am curious about the optimal epoch subtraction method.

You can specify any checkpoint selection you want but this is an unsupervised domain adaptation task, thus, by tryign to make the teacher optimal by any means w.r.t. the validation set is actually loosing the point. Re-iterating what I answered to you in a previous comment, for the teacher I just let it train for 100-120 epochs. And for the student I chose some with the best SI-SDR on libri1to3mix, or the best MOS or the best BAK on the validation set of Chime5. Again, in the names of the student files you can see the epoch and the method I used to pick the student checkpoints https://github.com/etzinis/unsup_speech_enh_adaptation/tree/main/pretrained_checkpoints.

> the lower the SISDR, the better the training

That is incorrect. The higher the SI-SDR metric, the better. 

> SISDR is negative

Again incorrect. SI-SDR is a metric of the signal fidelity that can take any real value. In our implementations we usually use a range between [-35, 35] or [-80, 80] which is usually caused by our numerical stability constant.

> Also, it used to display the mean, std, and median together, and I wonder what the purpose of displaying the three indicators together is.

If by "it" you refer to my script, plotting all metrics and not only the mean is very important to understand the behavior of your model and not simply try to beat some numbers. 

> I'm curious how the author checked the metrics to see which of the three metrics is meaningful, and did he consider mos or mean or std or median to choose the best epoch?

I have already answered to this question in your previous comment and this comment above. However, you correctly identify a problem that speech enhancement has in general, that is there is not a single metric that has to be optimized simply because human hearing in not 100% corresponding to any of the objective signal metrics out there. This is why people report PESQ, SI-SDR, BAK, MOS, etc. all together in order to show improvement in most or all of them.

Finally, you wrote a huge response without answering to my first response-comment which was the most important question to understand what kind of "different" and "not-reproducible" results you were getting. Anyhow, I think that helps.
